index.js

```diff
@@ -81,19 +105,25 @@ async function authorize(options) {
   };
 
   return fetch(`${ims}/ims/exchange/jwt/`, postOptions)
-    .then(res => res.json())
+    .catch(e => throwRequestFailedError(e.message))
+    .then(res => {
+      // if (res.ok) {
```

Comment:
Interestingly, even when the swap was successful, `res.ok` is `false` and the status code is `400`. I don't know why this is happening.